evalml/pipelines/components/transformers/imputers/imputer.py

```diff
@@ -158,15 +157,18 @@ def transform(self, X, y=None):
             return df
 
         X_no_all_null = X.ww.drop(self._all_null_cols)
+        original_schema = X_no_all_null.ww.schema
+        new_ltypes = None
 
         if self._numeric_cols is not None and len(self._numeric_cols) > 0:
             X_numeric = X.ww[self._numeric_cols.tolist()]
             imputed = self._numeric_imputer.transform(X_numeric)
-            for numeric_col in X_numeric.columns:
-                X_no_all_null.ww[numeric_col] = init_series(
-                    imputed[numeric_col],
-                    logical_type="Double",
-                )
+            X_no_all_null[X_numeric.columns] = imputed
+            # Numeric imputing may have changed logical types because of use of _get_new_logical_types_for_imputed_data
+            # --> this is only needed because of test_imputer_does_not_erase_ww_info test - otherwise we always assume ww types would be present
+            if imputed.ww.schema is None:
+                imputed.ww.init()
```

Comment:
Got it - so because we're doing a `ww.init()` call here, it does open up the possibility of reinferring and losing types if the simple imputer stopped maintaining the woodwork schema, but, like you say, there's not a great way to avoid this, and at least it would just be the numeric columns. The good thing is that we don't expect to actually need to init woodwork here, so we're okay leaving it as is.

And to answer your question: that is correct. We can fully maintain all the non integer types, and that allows us to not have any similar logic for any of the other types of columns.